# forms.py

from django import forms
from django.contrib.auth.models import User
from django.core.exceptions import ValidationError

class UserRegistrationForm(forms.ModelForm):
    username = forms.CharField(label='Nombre de Usuario', widget=forms.TextInput(attrs={'class': 'form-control font-weight-bold'}))
    email = forms.EmailField(label='Correo Electr칩nico', widget=forms.EmailInput(attrs={'class': 'form-control font-weight-bold'}))
    password = forms.CharField(label='Contrase침a', widget=forms.PasswordInput(attrs={'class': 'form-control font-weight-bold'}))
    password_confirm = forms.CharField(label='Confirmar Contrase침a', widget=forms.PasswordInput(attrs={'class': 'form-control font-weight-bold'}))

    class Meta:
        model = User
        fields = ('username', 'email', 'password')

    def clean_password_confirm(self):
        cd = self.cleaned_data
        if cd['password'] != cd['password_confirm']:
            raise ValidationError('Las contrase침as no coinciden.')
        return cd['password_confirm']

